yarn-project/noir-libs/noir-aztec/src/context.nr

```diff
@@ -287,7 +320,13 @@ impl PrivateContext {
         assert(item.public_inputs.call_context.msg_sender == self.inputs.call_context.storage_contract_address);
         assert(item.public_inputs.call_context.storage_contract_address == contract_address);
 
-        self.private_call_stack.push(item.hash());
+        let side_effect = SideEffectWithRange {
+            value: item.hash(),
+            start_side_effect_counter: self.inputs.side_effect_counter,
+            end_side_effect_counter: self.side_effect_counter,
+        };
+        self.private_call_stack.push(side_effect);
+        self.side_effect_counter++;
```

Comment:
Do we need to know here how many side effects the nested private call will have and increment the counter accordingly (instead of just by 1)?